### PR TITLE
Environment_Engine: adding replace space name method

### LINKED ISSUE #3214: Environment_Engine: Add ability to update panel connected space names
When working in a parametric workflow, it isn't currently easy to update space names once set if you want to include information that was only available after the panels were created.

@FelixMallinder-BH has prototyped a method which can allow this, taking in all the panels and checking if the connected spaces match any of the ones which need replacing and replacing them with suitable new names.

### PR DESCRIPTION
**Pull request by FelixMallinder-BH**
### Issues addressed by this PR

Closes #3214

### CODE REVIEW

File: Environment_Engine/Modify/ReplaceSpaceName.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using BH.oM.Base.Attributes;
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/ReplaceSpaceName.cs:1]: Please add a copyright header

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: Environment_Engine/Modify/ReplaceSpaceName.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            "This will go through every panels connected spaces and check to see if they are within the provided old space names. If they are, they will be updated with the corresponding item in the new space names matching the index of the old space names. Therefore it is vital to ensure the new space names order matches the order of the old space names.")]
+        [Input("panels", "The collection of Environment Panels to update space names for.")]
+        [Input("oldSpaceNames", "The old space names to replace for the panel.")]
+        [Input("newSpaceNames", "The replacement space names for the panel.")]
+        [Output("panel", "The panel with updated space names if the panel contained the old space name.")]
+        public static List<Panel> ReplaceSpaceName(this List<Panel> panels, List<string> oldSpaceNames, List<string> newSpaceNames)
```
Comment (FraserGreenroyd) [inline Environment_Engine/Modify/ReplaceSpaceName.cs:41]:
```suggestion
        public static List<Panel> ReplaceConnectedSpaceName(this List<Panel> panels, List<string> oldSpaceNames, List<string> newSpaceNames)
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by albinber — APPROVED** [review]
Helpful, well needed method. Looks good! Nice work @FelixMallinder-BH

**Comment by FraserGreenroyd** [comment]
> Helpful, well needed method. Looks good! Nice work @FelixMallinder-BH

You've been working with @kprusicka too much... 😉

**Review by FraserGreenroyd — APPROVED** [review]
Tested through on workshop for project with @FMHagley and @FelixMallinder-BH , works as expected. Approving to test further via alphas.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**